contrib/win32/win32compat/fileio.c

```diff
@@ -769,14 +769,24 @@ fileio_write(struct w32_io* pio, const void *buf, size_t max_bytes)
 int
 fileio_fstat(struct w32_io* pio, struct _stat64 *buf)
 {
-	int fd = _open_osfhandle((intptr_t)pio->handle, 0);
+	HANDLE dup_handle = 0;
+	if (!DuplicateHandle(GetCurrentProcess(), pio->handle, GetCurrentProcess(), &dup_handle, 0,
```

Comment:
I don't think we need to duplicate handle here..
when you do _open_osfhandle it will return the c run-time file descriptor. If you close the c run-time file descriptor then it shouldn't have any impact on the os handle (pio->handle).

Reply:
According to the docs for _open_osfhandle, _close will close the underlying handle:
"To close a file opened with , call _close. The underlying handle is also closed by a call to _close, so it is not necessary to call the Win32 function CloseHandle on the original handle"

Though I'm not an expert in these APIs.

Reply:
My bad, I didn't read the msdn documentation properly.. You are right.